R/database.R

```diff
@@ -2628,3 +2630,385 @@ get_gene_expression = function(metadata,
   }
   return(expression_wider)
 }
+
+
+#' Load the manta output for a set of samples
+#'
+#' This is a convenience wrapper function for get_manta_sv_by_sample. See that function for more information.
+#'
+#' @param these_samples_metadata The only required parameter is a metadata table (data frame) that must contain a row for each sample you want the data from. The additional columns the data frame needs to contain, besides sample_id, are: unix_group, genome_build, seq_type, pairing_status
+#' @param min_vaf The minimum tumour VAF for a SV to be returned.
+#' @param min_score The lowest Manta somatic score for a SV to be returned.
+#' @param pass If set to TRUE, include SVs that are annotated with PASS in FILTER column. Default is TRUE.
+#' @param with_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses).
+#' @param from_flatfile Set to TRUE by default.
+#' @param projection The projection genome build.
+#' @param all_sv_grch37 Optional data frame with merged manta results from get_manta_sv(), useful for when individual sample in a specific genome projection is not available. If not provided, this function will run get_manta_sv(projection = "grch37").
+#' @param all_sv_hg83 Optional data frame with merged manta results from get_manta_sv(), useful for when individual sample in a specific genome projection is not available. If not provided, this function will run get_manta_sv(projection = "hg38").
+#'
+#' @return a data frame containing the Manta outputs from all sample_id in these_samples_metadata in a bedpe-like format with additional columns extracted from the VCF column.
+#' @import tidyverse
+#' @export
+#'
+#' @examples
+#' my_samples = c("00-20702T", "00-15201_tumorB")
+#'
+#' multi_meta = get_gambl_metadata() %>% dplyr::filter(sample_id %in% my_samples)
+#'
+#' my_samples_sv = get_manta_sv_by_samples(these_samples_metadata = multi_meta)
+#'
+get_manta_sv_by_samples = function(these_samples_metadata,
+                                   this_sample_id,
+                                   min_vaf = 0.1,
+                                   min_score = 40,
+                                   pass = TRUE,
+                                   with_chr_prefix = FALSE,
+                                   from_flatfile = TRUE,
+                                   projection = "grch37",
+                                   all_sv_grch37,
+                                   all_sv_hg38){
+
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+
+  #load merged files if not provided
```

Comment:
I see no reason to make these parameters. Why wouldn't this function just load these files automatically? It only needs to happen once, right? It's confusing to have these new parameters and unnecessary confusion if they are rarely (or never) going to be used

Reply:
For this function, I have removed the two parameters `all_sv_grch37` and `all_sv_hg38` and replaced with a new optional parameter called, `all_sv`. If not provided, the function will get it for you, based on the selected genome projection.

Reply:
Is this resolved?